evmbin/src/main.rs

```diff
@@ -89,6 +89,9 @@ State test options:
 General options:
     --json             Display verbose results in JSON.
     --std-json         Display results in standardized JSON format.
+    --dump-json        Display result state dump in standardized JSON format.
+    --err-only         With --std-json redirect to err output only.
+    --out-only         With --std-json redirect to out output only.
```

Comment:
Maybe this is a dumb idea, but there's a risk users get it wrong and use `--err-only` without `--dump-json` so we could do

```
--dump-json … …
--dump-json-err-only … …
--dump-json-out-only … …
```
i.e. have the three options mutually exclusive instead of combining.